src/mame/seta/speglsht.cpp

```diff
@@ -376,23 +372,23 @@ void speglsht_state::machine_reset()
 void speglsht_state::video_start()
 {
 	m_bitmap = std::make_unique<bitmap_ind16>(512, 512);
-}
-
-#define PLOT_PIXEL_RGB(x,y,r,g,b)   if(y>=0 && x>=0 && x<512 && y<512) \
-{ \
-		bitmap.pix(y, x) = (b) | ((g)<<8) | ((r)<<16); \
+	save_item(NAME(m_videoreg));
 }
 
 uint32_t speglsht_state::screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect)
 {
-	int dy=(m_videoreg&0x20)?(256*512):0; //visible frame
+	int const dy = BIT(m_videoreg, 5) ? (256 * 512) : 0; //visible frame
 
-	for(int y=0;y<256;y++)
+	for (int y = 0; y < 256; y++)
 	{
-		for(int x=0;x<512;x++)
+		int const dsty = y - 5;
+		uint32_t *const dstline = &bitmap.pix(dsty);
+		for (int x = 0; x < 512; x++)
 		{
-			int tmp=dy+y*512+x;
-			PLOT_PIXEL_RGB(x-67,y-5,(m_framebuffer[tmp]>>0)&0xff,(m_framebuffer[tmp]>>8)&0xff,(m_framebuffer[tmp]>>16)&0xff);
+			int const tmp = dy + y * 512 + x;
+			int const dstx = x - 67;
+			if (cliprect.contains(dstx, dsty))
+				dstline[dstx] = rgb_t((m_framebuffer[tmp] >> 0) & 0xff, (m_framebuffer[tmp] >> 8) & 0xff, (m_framebuffer[tmp] >> 16) & 0xff);
 		}
 	}
```

Comment:
Can you rework this to loop over the clip rectangle and calculate the source location from that instead?  Something like:
```c++
	int y = cliprect.top() + 5 + (BIT(m_videoreg, 5) ? (256 * 512) : 0);
	for (int dsty = cliprect.top(); dsty <= cliprect.bottom(); dsty++, y += 512)
	{
		uint32_t *const dstline = &bitmap.pix(dsty);
		for(int dstx = cliprect.left(); dstx <= cliprect.right(); dstx++)
		{
			uint32_t const pix = m_framebuffer[y + dstx + 67];
			dstline[dstx] = rgb_t((pix >> 0) & 0xff, (pix >> 8) & 0xff, (pix >> 16) & 0xff);
		}
	}
```

That gets the multiplications and extra tests out of the inner loop.